src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -326,6 +326,28 @@ protected function createFormBuilder($data = null, array $options = array()): Fo
         return $this->container->get('form.factory')->createBuilder(FormType::class, $data, $options);
     }
 
+    /**
+     * Handles request and check form validity.
+     *
+     * @final
+     */
+    protected function isFormValid(FormInterface $form, Request $request = null): bool
+    {
+        if (!$request) {
+            $request = $this->container->get('request_stack')->getCurrentRequest();
+        }
+
+        if (!$request) {
+            throw new \LogicException('You must pass a request as second argument because the request stack was empty.');
+        }
+
+        if ($form->isSubmitted()) {
```

Comment:
can be the first check

Reply:
What about:
```php
if (!$form->isSubmitted()) {
    $form->handleRequest($request);
}

return $form->isSubmitted() && $form->isValid();
```
?
Throwing the exception here does not make sense to me due to the name of the shortcut, and given that the form would already do it by itself.

Reply:
then we conditionally ignore $request which could make this shortcut more confusing. IMHO it's not meant to be used once the form is already handled.

Reply:
@ro0NL I updated the PR ; thanks

Reply:
@ro0NL 
>IMHO it's not meant to be used once the form is already handled.

It may, think about `forward` call(s).

Reply:
really? That means the form instance is passed to the sub-request, to be validated twice.. :thinking:

Reply:
Actually it can be recreated and rehandled through this short cut yes, otherwise we should not care about having this clause guard in the first place, but people can do weird things with forward call like chaining or multiple calls during the same request.
So given the name of the shortcut, and unless we can't get it more explicit that it mutates the form, we should make it silent here and just return a bool, IMHO that would avoid WTF moments.

Reply:
> otherwise we should not care about having this clause guard in the first place

hm, that is actually required to do so... as the "default crash" might not always happen (depending on the submitted data in the request):

https://github.com/symfony/symfony/blob/f9414a8e818988ebaff4f6ec219cccce811f06c2/src/Symfony/Component/Form/Form.php#L502-L506

Also im i think we should use a different name indeed. What about `submitForm(...): bool`?

Reply:
actually no :) the check is fine. But intend should be clarified by name (`submitForm()`).

Removing it would cause to return a possible previous state.